### PR TITLE
Do not rely on isinstance to find the default engine.

### PR DESCRIPTION
**Pull request by apollo13**
This allows django_coverage_plugin to work again. (I still have to test django_coverage_plugin ;))

### CODE REVIEW

**Comment by apollo13** [comment]
coverage output:
```
py35-djangotip create: /home/florian/sources/django_coverage_plugin/.tox/py35-djangotip
py35-djangotip installdeps: unittest-mixins==1.1, https://github.com/apollo13/django/archive/django_coverage.tar.gz
py35-djangotip develop-inst: /home/florian/sources/django_coverage_plugin
py35-djangotip installed: coverage==4.2,Django==1.10rc1,-e git+https://github.com/nedbat/django_coverage_plugin.git@6ca652519934cab7fa248f02e6197d0a459d90a9#egg=django_coverage_plugin,six==1.10.0,unittest-mixins==1.1
py35-djangotip runtests: PYTHONHASHSEED='3905908550'
py35-djangotip runtests: commands[0] | python -c import tests.banner
CPython 3.5.2+; Django 1.11.dev20161007122508
py35-djangotip runtests: commands[1] | python -m unittest discover -b
............ss.........................................
----------------------------------------------------------------------
Ran 55 tests in 0.370s

OK (skipped=2)
__________________________________________________________________________________________________________________ summary ___________________________________________________________________________________________________________________
  py35-djangotip: commands succeeded
  congratulations :)
```

@nedbat Can you verify that against the coverage plugin? I just did run the testsuite…

File: django/template/engine.py [hunk 1]
```diff
@@ -68,16 +69,11 @@ def get_default():
...
         """
-        # Since Engine is imported in django.template and since
-        # DjangoTemplates is a wrapper around this Engine class,
-        # local imports are required to avoid import loops.
-        from django.template import engines
-        from django.template.backends.django import DjangoTemplates
-        django_engines = [engine for engine in engines.all()
-                          if isinstance(engine, DjangoTemplates)]
+        django_engines = [engine for engine in engines.templates.values()
+                          if engine['BACKEND'] == 'django.template.backends.django.DjangoTemplates']
```
Comment (apollo13) [inline django/template/engine.py]: Those are not ment to be considered as default, after all you couldn't "subclass" Django template engine before the multiple template engine stuff either…

**Comment by timgraham** [comment]
Florian, what's the status of this?

**Comment by apollo13** [comment]
waiting on @nedbat for feedback on whether that fixes django-template-coverage or not

**Comment by nedbat** [comment]
Sadly, I haven't had time to work on the Django template coverage plugin in quite a while.  I'm not sure how long it will be before it comes to the top of the list again.  If you know someone who wants to take on a project, let me know.  Sorry to drop the ball... :(

**Comment by PamelaM** [comment]
I just submitted a PR to django_coverage_plugin that solves the problem for dj-cov-plg (https://github.com/nedbat/django_coverage_plugin/pull/30) w/o requiring a fix in Django.  However, this PR would also have resolved the failing test.

**Comment by apollo13** [comment]
Perfect, closing here.

**Pull request closed without merging**